pkg/client/restclient/request.go

```diff
@@ -105,16 +106,13 @@ type Request struct {
 	resource     string
 	resourceName string
 	subresource  string
-	selector     labels.Selector
 	timeout      time.Duration
 
 	// output
 	err  error
 	body io.Reader
 
-	// The constructed request and the response
-	req  *http.Request
-	resp *http.Response
+	ctx context.Context
```

Comment:
The API that takes a `context.Context` should be clearly document what can happen with `context.Context` as far as tracing, auth, and other "magic." The short answer is if you don't want those things, don't use those things. I still think it's powerful with regards to cancellation, timeouts, and deadlines, especially since it's now part of the standard library and integrated into other standard libraries.

The main argument against re-using the timeout parameter is that it changes existing client behavior.